ingest/writer_functions.py

```diff
@@ -0,0 +1,203 @@
+from __future__ import annotations
+from typing import TextIO
+import re
+import os
+import gzip
+import json
+
+try:
+    from ingest_files import IngestFiles
+except ImportError:
+    from .ingest_files import IngestFiles
+
+# regex to split on commas and tabs
+COMMA_OR_TAB = r"[,\t]"
+
+# regex to split on comma/tab/space
+ALL_DELIM = r"[\s,\t]"
+
+# Gzip magic number
+GZIP_MAGIC_NUM = b'\x1f\x8b'
+
+ALLOWED_FILE_TYPES = ["text/csv", "text/plain", "text/tab-separated-values"]
+
+def is_gz_file(filepath) -> bool:
+    """
+    Determine if a file is gzipped by checking first two bytes against GZIP_MAGIC_NUM
+
+    :param filepath: (str) path to file to check
+    :returns: (bool)
+    """
+    with open(filepath, 'rb') as test_f:
+        return test_f.read(2) == GZIP_MAGIC_NUM
```

Comment:
Let's use the [same exact name as core JS and Ruby](https://github.com/broadinstitute/single_cell_portal_core/search?q=gzip+magic+number), for good measure.

Supplemental notes:
* The fact that [JS only checks 1 byte](https://github.com/broadinstitute/single_cell_portal_core/blob/762ebd8fe9824a228e5133f99ef856e29877238c/app/javascript/lib/validation/validate-file-content.js#L323) and Ruby and this Python check 2 bytes makes me wary.
* Can this function replace the [equivalent function in ingest_files.py](https://github.com/broadinstitute/scp-ingest-pipeline/blob/913bf2b3f11c104bbbcddce0be4d2f6ed6be8859/ingest/ingest_files.py#L103)?

```suggestion
# The signature first byte of file compressed via gzip
GZIP_MAGIC_NUMBER = b'\x1f\x8b'

ALLOWED_FILE_TYPES = ["text/csv", "text/plain", "text/tab-separated-values"]

def is_gz_file(filepath) -> bool:
    """
    Determine if a file is gzipped by checking first two bytes against GZIP_MAGIC_NUMBER

    :param filepath: (str) path to file to check
    :returns: (bool)
    """
    with open(filepath, 'rb') as test_f:
        return test_f.read(2) == GZIP_MAGIC_NUMBER
```